MAUI/TreeView/mvvm.md

```diff
@@ -0,0 +1,397 @@
+        australia.States = new ObservableCollection<Countries>();
+        australia.States.Add(nsw);
+        australia.States.Add(victoria);
+        nsw.States = new ObservableCollection<Countries>();
+        nsw.States.Add(sydney);
+        var usa = new Countries() { Name = "United States of America" };
+        var california = new Countries() { Name = "California" };
+        usa.States = new ObservableCollection<Countries>();
+        usa.States.Add(california);
+      
+        this.CountriesInfo = new ObservableCollection<Countries>();
+        CountriesInfo.Add(australia);
+        CountriesInfo.Add(usa);
+
+        SelectedCountries = new ObservableCollection<object>();
+        SelectedCountries.Add(nsw);
+        SelectedCountries.Add(victoria);
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Commands
+
+### Tap command
+
+The `TapCommand` will be triggered whenever tapping the item and passing the `TreeViewNode` as command parameter.
+
+{% tabs %}
+{% highlight c# tabtitle="CommandViewModel.cs" %}
+
+treeView.TapCommand = viewModel.TappedCommand;
+
+public class CommandViewModel
+{
+    private Command<object> tappedCommand;
+
+    public Command<object> TappedCommand
+    {
+        get { return tappedCommand; }
+        set { tappedCommand = value; }
+    }
+
+    public CommandViewModel()
+    {            
+        TappedCommand = new Command<object>(TappedCommandMethod);
+    }
+
+    private void TappedCommandMethod(object obj)
+    {
+        App.Current.MainPage.DisplayAlert("Alert", ((obj as TreeViewNode).Content as FileManager).ItemName+ "  is Tapped","OK");
+    }   
+}
+{% endhighlight %}
+{% endtabs %}
+
+### LongPressCommand
+
+The `LongPressCommand` will be triggered whenever an item is long pressed and passing the `TreeViewNode` as command parameter.
+
+{% tabs %}
+{% highlight c# tabtitle="CommandViewModel.cs" %}
+
+treeView.LongPressCommand = viewModel.LongPressCommand;
+
+public class CommandViewModel
+{
+    private Command<Object> longPressCommand;
+
+    public Command<object> LongPressCommand
+    {
+        get { return longPressCommand; }
+        set { longPressCommand = value; }
+    }
+
+    public CommandViewModel()
+    {            
+        LongPressCommand = new Command<object>(LongPressCommandMethod);
+    }
+
+    private void LongPressCommandMethod(object obj)
+    {          
+        App.Current.MainPage.DisplayAlert("Alert", ((obj as TreeViewNode).Content as FileManager).ItemName+ "  is LongPressed","OK");
+    }   
+}
+{% endhighlight %}
+{% endtabs %}
+
+### Expand command
+
+The `ExpandCommand` will be triggered while expanding the node and passing the `TreeViewNode` as command parameter. TreeView expands the node based on the return value of `CanExecute` method implementation of `ExpandCommand`. If you return false, then expand action will be canceled. Execute method implementation of `ExpandCommand` will get called after expanding of node.
```

Comment:
The `ExpandCommand` will be triggered while expanding the node and passing the `TreeViewNode` as a command parameter. TreeView expands the node based on the return value of the `CanExecute` method implementation of `ExpandCommand`. If you return false, then expand action will be canceled. Execute method implementation of `ExpandCommand` will get called after expanding the node.

Reply:
Replaced with the above content in Expand command